app/models/manageiq/providers/openshift/container_manager/refresher.rb

```diff
@@ -2,21 +2,20 @@ module ManageIQ::Providers
   module Openshift
     class ContainerManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
       include ::EmsRefresh::Refreshers::EmsRefresherMixin
+      include ::EmsRefresh::Refreshers::ContainerRefresherMixin
 
       KUBERNETES_EMS_TYPE = ManageIQ::Providers::Kubernetes::ContainerManager.ems_type
 
-      def self.entities
-        [{:name => 'routes'}, {:name => 'projects'}]
-      end
-
       def parse_inventory(ems, _targets = nil)
-        openshift_entities = ems.with_provider_connection { |client| fetch_entities(client, self.class.entities) }
-        kubernetes_entities = ems.with_provider_connection(:service => KUBERNETES_EMS_TYPE) do |client|
-          fetch_entities(client, ManageIQ::Providers::Kubernetes::ContainerManager::Refresher.entities)
+        entities = ems.with_provider_connection(:service => KUBERNETES_EMS_TYPE) do |kubeclient|
+          ems.with_provider_connection do |openshift_client|
+            openshift_fetch_inventory(openshift_client).merge(
+              kubernetes_fetch_inventory(kubeclient)
+            )
+          end
```

Comment:
This block reads a bit confusing.  It looks like it should just be:

```ruby
kube_entities = ems.with_provider_connection(:service => KUBERNETES_EMS_TYPE) do |kubeclient|
  kubernetes_fetch_inventory(kubeclient)
end
openshift_entities = ems.with_provider_connection do |openshift_client|
  openshift_fetch_entities(openshift_client)
end
entities = openshift_entities.merge(kube_entities)
```

Nesting the two `with_provider_connection` calls makes it look like one connection depends on the other.  When, really it's just two calls in serial.